README.md

```diff
@@ -187,21 +179,32 @@ export const load = async (csvPath: string, column: string) => {
   // Extract the selected column from the CSV file
   const documents = data.map((row) => row[column] as string);
 
-  // Create a Pinecone index with the name "word-embeddings" and a dimension of 384
-  await createIndexIfNotExists(pineconeClient, indexName, 384);
+  // Get index name
+  const indexName = getEnv("PINECONE_INDEX");
+
+  // Check whether the index already exists. If it doesn't, create 
+  // a Pinecone index with a dimension of 384 to hold the outputs
+  // of our embeddings model.
+  const indexList = await pinecone.listIndexes();
+  if (indexList.indexOf({ name: indexName }) === -1) {
+    await pinecone.createIndex({ name: indexName, dimension: 384, waitUntilReady: true })
+  }
 
-  // Select the target Pinecone index
-  const index = pineconeClient.Index(indexName);
+  // Select the target Pinecone index. Passing the TextMetadata generic type parameter
+  // allows typescript to know what shape to expect when interacting with a record's
+  // metadata field without the need for additional type casting.
+  const index = pinecone.index<TextMetadata>(indexName);
 
   // Start the progress bar
   progressBar.start(documents.length, 0);
 
   // Start the batch embedding process
   await embedder.init();
-  await embedder.embedBatch(documents, 1, async (embeddings) => {
+  await embedder.embedBatch(documents, 100, async (embeddings) => {
```

Comment:
I can change this number back. I was experimenting with different values here and it seems like changing this number didn't have much impact on overall load time.

Reply:
Seems fine to me. When I ran through this example during the hiring process one of these steps (I believe it was the embedding / upserting) took an extremely long time. I'm not sure if that was something on my end or something with how the example was handling things.

I would have assumed it was this bit here though, yeah: https://github.com/pinecone-io/semantic-search-example/blob/00faff1319d0fab83f87490181284cdf6e5a036b/src/load.ts#L49